hw/ip/pattgen/data/pattgen_testplan.hjson

```diff
@@ -81,4 +81,16 @@
       tests: ["pattgen_stress_all"]
     }
   ]
+
+  covergroups: [
+    {
+      name: pattgen_op_cg
+      desc: '''
+            Covers the following conditions:
+              - whether single or dual channel are enabled
+              - whether channel polarity bit is set/unset
+              - whether channel are run at high/low throughput
+            '''
+    }
```

Comment:
we can consider to cover every bit of data and predivider are toggled. use auto bins can do it easily. there are some examples [here](https://verificationguide.com/systemverilog/systemverilog-functional-coverage/)

and also cover some value of data length and number of repetitions.

Looks like we should have a review meeting to go over these.

Reply:
+1 
There are several things about pattgen that can be captured in functional coverage plan. Please work with the designer to come up with a detailed plan. We should have ideally documented it before V1.